### ISSUE TITLE
pyproj 2.5.0 does not convert from epsg:4326 to epsg:4267

### ISSUE THREAD

**Issue by aleksanderczeszyk-intive** [pyproj4/pyproj#571]
Hi guys,
I may have found a problem with latest version of pyproj. I am designing test cases that involve converting geojson geometries from epsg:4326 to epsg:4267. The example GeoJson polygon is here:
```
{
            "type": "Polygon",
            "coordinates": [
                [
                    [
                        32.794189453125,
                        3.995780512963038
                    ],
                    [
                        30.064086914062496,
                        3.5572827265412794
                    ],
                    [
                        29.086303710937504,
                        0.4394488164139768
                    ],
                    [
                        33.013916015625,
                        -0.42296979098860243
                    ],
                    [
                        35.0079345703125,
                        0.6921218386632358
                    ],
                    [
                        34.8101806640625,
                        3.2556928748506118
                    ],
                    [
                        32.794189453125,
                        3.995780512963038
                    ]
                ]
            ]
        }
```
After that the polygons are converted to WKT. The result for pyproj 2.4.2 looks like this:

```
('POLYGON ((32.794189453125 3.995780512963038, 30.064086914062496 '
 '3.5572827265412794, 29.086303710937504 0.4394488164139768, 33.013916015625 '
 '-0.42296979098860243, 35.0079345703125 0.6921218386632358, 34.8101806640625 '
 '3.2556928748506118, 32.794189453125 3.995780512963038))')
('POLYGON ((32.79472783719299 3.9940903934167866, 30.064484089015565 '
 '3.5556408566034987, 29.086611305198623 0.4378252001668296, 33.01440476269899 '
 '-0.4246599023882488, 35.00852518517081 0.6903896943025318, '
 '34.810799463983386 3.253962650695694, 32.79472783719299 3.9940903934167866))')
```

The first polygon is epsg:4326 and second epsg:4267.

However after updating pyproj the transformation does not occur. conversion to epsg:4267 does not make any effect and the polygon stays the same. The method used for conversion looks like this:
```
    def convert_from_epsg_4326_to_different_projection(self, geometry, projection):
        new_geometry = copy.deepcopy(geometry)
        epsg_4326 = pyproj.Proj("epsg:4326")
        new_projection = pyproj.Proj(projection, preserve_units=True)
        if new_geometry["type"] == "Point":
            new_geometry["coordinates"] = list(pyproj.transform(
                epsg_4326,
                new_projection,
                new_geometry["coordinates"][0],
                new_geometry["coordinates"][1]
            ))
        elif new_geometry["type"] == "LineString":
            for point_id in range(len(new_geometry["coordinates"])):
                new_geometry["coordinates"][point_id] = list(pyproj.transform(
                    epsg_4326,
                    new_projection,
                    new_geometry["coordinates"][point_id][0],
                    new_geometry["coordinates"][point_id][1]
                ))
        elif new_geometry["type"] == "Polygon":
            for point_id in range(len(new_geometry["coordinates"][0])):
                new_geometry["coordinates"][0][point_id] = list(pyproj.transform(
                    epsg_4326,
                    new_projection,
                    new_geometry["coordinates"][0][point_id][0],
                    new_geometry["coordinates"][0][point_id][1]
                ))

        return new_geometry
```
Is my thinking right? is that an issue or maybe 2.4.2 did something it shouldn't?
Thanks in advance

**Comment by snowman2** [comment]
Mind adding the output of `python -m pyproj -v`? Also, how did you install pyproj?

**Comment by aleksanderczeszyk-intive** [comment]
For 2.4.2:
```
pyproj info:
    pyproj: 2.4.2
      PROJ: 6.2.1
  data dir: /Users/Czeszyk/Projects/oden_e2e/oden_e2e_venv/lib/python3.6/site-packages/pyproj/proj_dir/share/proj

System:
    python: 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 05:52:31)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]
executable: /Users/Czeszyk/Projects/oden_e2e/oden_e2e_venv/bin/python
   machine: Darwin-19.3.0-x86_64-i386-64bit

Python deps:
       pip: 19.3.1
setuptools: 39.0.1
    Cython: None
```
For 2.5.0
```
pyproj info:
    pyproj: 2.5.0
      PROJ: 6.3.1
  data dir: /Users/Czeszyk/Projects/oden_e2e/oden_e2e_venv/lib/python3.6/site-packages/pyproj/proj_dir/share/proj

System:
    python: 3.6.5 (v3.6.5:f59c0932b4, Mar 28 2018, 05:52:31)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]
executable: /Users/Czeszyk/Projects/oden_e2e/oden_e2e_venv/bin/python
   machine: Darwin-19.3.0-x86_64-i386-64bit

Python deps:
       pip: 19.3.1
setuptools: 39.0.1
    Cython: None
```
Installed using pip.

**Comment by jorisvandenbossche** [comment]
Can you try with a smaller reproducer? Eg I extracted:

```
import pyproj 
    
epsg_4326 = pyproj.Proj("epsg:4326")
new_projection = pyproj.Proj("epsg:4267", preserve_units=True) 
pyproj.transform(epsg_4326, new_projection, 32.794189453125, 3.995780512963038) 
```

which outputs the same with pyproj master

**Comment by aleksanderczeszyk-intive** [comment]
Yep, tried out in the console:
```
>>> pyproj.transform(epsg_4326, new_projection, 32.794189453125, 3.995780512963038) 
(32.794189453125, 3.995780512963038)
```

**Comment by snowman2** [comment]
This is a PROJ 6.3.1 issue. Looks like it is fixed in 7.0.0:
```
(proj631) $ echo 32.794189453125 3.995780512963038 | cs2cs -f %.12f "epsg:4326" "epsg:4267"
32.794189453125	3.995780512963 0.000000000000
(proj621) $ echo 32.794189453125 3.995780512963038 | cs2cs -f %.12f "epsg:4326" "epsg:4267"
32.794727837193	3.994090393417 0.000000000000
(proj700) $echo 32.794189453125 3.995780512963038 | cs2cs -f %.12f "epsg:4326" "epsg:4267"
32.794189453125	3.995780512963 0.000000000000
```

This will be addressed in the next release as PROJ 7.0.0 will be in the wheels.

**Comment by snowman2** [comment]
Actually, nevermind - I mixed the order and got things confused. Looks like it is still an issue. I will forward.
